### PR TITLE
Update the documentation at "README.adoc"

### LINKED ISSUE #121: Update the documentation at "README.adoc"
### Describe your use-case which is not covered by existing documentation.

The current documentation we have at https://github.com/jenkins-infra/docs.jenkins.io/blob/main/README.adoc is simply a stub, and needs reworking to provide more info such as the separation of the versioned controlled Antora part and the non-versioned controlled Gatsby part, where the latter is a basis of the to-be-replaced https://jenkins.io/ website. 

Also, we could add some link to the existing Gatsby "README.md" at https://github.com/jenkins-infra/docs.jenkins.io/tree/main/site#readme. 

### Reference any relevant documentation, other materials or issues/pull requests that can be used for inspiration.

_No response_

### PR DESCRIPTION
**Pull request by Brijeshthummar02**
Fixes #121

### CODE REVIEW

**Review by krisstern — CHANGES_REQUESTED** [review]
My apologies @Brijeshthummar02, the README is actually and AsciiDoc, so you will need to revert to the syntax used previously.

**Comment by Brijeshthummar02** [comment]
can you specify the change i'm not geting it.
@krisstern

**Comment by krisstern** [comment]
@Brijeshthummar02 Do you know the differences between an `.adoc` document and a `.md` document?

**Comment by Brijeshthummar02** [comment]
@krisstern i searched it, i'm working on it, submitting update soon.

File: README.adoc [hunk 1]
```diff
@@ -1,47 +1,73 @@
...
+    ```
+
+Once the site is up and running, it should be available locally at:
+
+- [http://127.0.0.1:8080](http://127.0.0.1:8080)
+
+## Documentation Structure
+
+- **Antora Documentation**: This version-controlled documentation covers different guides (e.g., Developer Guide, User Guide).
+- **Gatsby Documentation**: Non-versioned content managed via Gatsby. This forms the basis of the current [https://jenkins.io/](https://jenkins.io/) site, which will be replaced in the future. For more details on how to work with the Gatsby site, refer to its dedicated README at:
```
Comment (krisstern) [inline README.adoc:68]:
```suggestion
* **Gatsby Documentation**: Non-versioned content managed via Gatsby. This forms the basis of the current link:https://jenkins.io/[https://jenkins.io/] site, which will be replaced in the future. For more details on how to work with the Gatsby site, refer to its dedicated README documentation at: 
```

File: README.adoc [hunk 2]
```diff
@@ -1,47 +1,73 @@
...
+    make all
+    ```
+
+Once the site is up and running, it should be available locally at:
+
+- [http://127.0.0.1:8080](http://127.0.0.1:8080)
+
+## Documentation Structure
+
+- **Antora Documentation**: This version-controlled documentation covers different guides (e.g., Developer Guide, User Guide).
```
Comment (krisstern) [inline README.adoc:67]:
```suggestion
* **Antora Documentation**: This version-controlled documentation covers different guides (e.g., Developer Guide, User Guide).
```

File: README.adoc [hunk 3]
```diff
@@ -1,47 +1,73 @@
...
+
+### Prerequisites
+
+1. **Install Node.js and npm**
+
+It is recommended to install Node.js and npm using the Node Version Manager (nvm):
+- NVM repository: [https://github.com/nvm-sh/nvm](https://github.com/nvm-sh/nvm)
+
+Alternatively, you can download and install Node.js from the official Node.js website:
+- Node.js official website: [https://nodejs.org](https://nodejs.org)
```
Comment (krisstern) [inline README.adoc:28]:
```suggestion
* Node.js official website: [https://nodejs.org](https://nodejs.org)
```

File: README.adoc [hunk 4]
```diff
@@ -1,47 +1,73 @@
...
+2. **Non-versioned documentation** (using Gatsby for rendering the current website)
+
+This guide focuses on building the version-controlled documentation using Antora.
+
+### Prerequisites
+
+1. **Install Node.js and npm**
+
+It is recommended to install Node.js and npm using the Node Version Manager (nvm):
+- NVM repository: [https://github.com/nvm-sh/nvm](https://github.com/nvm-sh/nvm)
```
Comment (krisstern) [inline README.adoc:25]:
```suggestion
* NVM repository: [https://github.com/nvm-sh/nvm](https://github.com/nvm-sh/nvm)
```

File: README.adoc [hunk 5]
```diff
@@ -1,47 +1,73 @@
...
+
+Once the site is up and running, it should be available locally at:
+
+- [http://127.0.0.1:8080](http://127.0.0.1:8080)
+
+## Documentation Structure
+
+- **Antora Documentation**: This version-controlled documentation covers different guides (e.g., Developer Guide, User Guide).
+- **Gatsby Documentation**: Non-versioned content managed via Gatsby. This forms the basis of the current [https://jenkins.io/](https://jenkins.io/) site, which will be replaced in the future. For more details on how to work with the Gatsby site, refer to its dedicated README at: 
+  [https://github.com/jenkins-infra/docs.jenkins.io/tree/main/site#readme](https://github.com/jenkins-infra/docs.jenkins.io/tree/main/site#readme)
```
Comment (krisstern) [inline README.adoc:69]:
```suggestion
** link:https://github.com/jenkins-infra/docs.jenkins.io/tree/main/site#readme/[https://github.com/jenkins-infra/docs.jenkins.io/tree/main/site#readme/]
```

File: README.adoc [hunk 6]
```diff
@@ -1,47 +1,73 @@
...
+
+## Contributing
+
+If you have any contributions or updates for the documentation, feel free to submit a pull request.
```
Comment (krisstern) [inline README.adoc:73]:
```suggestion
If you have any queries or contributions or updates for the documentation, feel free to create an issue and/or submit a pull request.
```

**Review by krisstern — CHANGES_REQUESTED** [review]

**Comment by Brijeshthummar02** [comment]
@krisstern is latest commit fine?

**Comment by krisstern** [comment]
Hi @Brijeshthummar02 I don't see any changes from you. You can simply commit my suggestions in a single batch by going to https://github.com/jenkins-infra/docs.jenkins.io/pull/135/files.

File: README.adoc [hunk 7]
```diff
@@ -1,47 +1,78 @@
...
+[source,bash]  
 ----
 make all
 ----
-+
-Once the Antora site is up and running it should be available locally at link:http://127.0.0.1:8080/[http://127.0.0.1:8080]. 
+
+Once the site is up and running, it should be available locally at:
+
+* link:http://127.0.0.1:8080/[http://127.0.0.1:8080]
```
Comment (krisstern) [inline README.adoc:70]:
```suggestion
* link:http://127.0.0.1:8080/[http://127.0.0.1:8080]

```

File: README.adoc [hunk 8]
```diff
@@ -1,47 +1,78 @@
...
+== Documentation Structure
+
+* **Antora Documentation**: This version-controlled documentation covers different guides (e.g., Developer Guide, User Guide).
+* **Gatsby Documentation**: Non-versioned content managed via Gatsby. This forms the basis of the current link:https://jenkins.io/[https://jenkins.io/] site, which will be replaced in the future. For more details on how to work with the Gatsby site, refer to its dedicated README documentation at:
+** link:https://github.com/jenkins-infra/docs.jenkins.io/tree/main/site#readme/[https://github.com/jenkins-infra/docs.jenkins.io/tree/main/site#readme/][https://github.com/jenkins-infra/docs.jenkins.io/tree/main/site#readme/]
+If you have any contributions or updates for the documentation, feel free to submit a pull request.
```
Comment (krisstern) [inline README.adoc:76]:
```suggestion

```

**Review by krisstern — CHANGES_REQUESTED** [review]

File: README.adoc [hunk 9]
```diff
@@ -1,47 +1,78 @@
...
+== Contributing
```
Comment (krisstern) [inline README.adoc:77]:
```suggestion
== Contributing

```

**Review by krisstern — CHANGES_REQUESTED** [review]

**Comment by Brijeshthummar02** [comment]
@krisstern  is it good to go now?

File: README.adoc [hunk 10]
```diff
@@ -1,47 +1,78 @@
...
-npm i
+npm install
 ----
 
-. **Run the Makefile**
-+
-Run the following at the root of the repository:
-+
-[source,bash]
+=== Building the Docs
```
Comment (krisstern) [inline README.adoc:59]:
```suggestion
=== Building the Versioned Docs
```

File: README.adoc [hunk 11]
```diff
@@ -1,47 +1,78 @@
...
++
+Alternatively, you can download and install Node.js from the official Node.js website:
+* Node.js official website: [https://nodejs.org](https://nodejs.org)
 +
-** Alternatively, download and install Node.js from the official Node.js website: link:https://nodejs.org/[https://nodejs.org]
-** After installing via either way, verify the installation by running the following commands in your terminal or command prompt:
+After installation, verify that both `node` and `npm` are correctly installed:
 +
-[source,bash]
+[source,bash]
```
Comment (krisstern) [inline README.adoc:35]:
```suggestion
[source,bash]
```

File: README.adoc [hunk 12]
```diff
@@ -1,47 +1,78 @@
...
 
-[start=1]
+[start=1]  
+. **Version-controlled documentation** (managed using Antora) 
+
+. **Non-versioned documentation** (using Gatsby for rendering the current website)
+
+This guide focuses on building the version-controlled documentation using Antora.
+
+=== Prerequisites
```
Comment (krisstern) [inline README.adoc:22]:
```suggestion
=== Installing the Prerequisites
```

File: README.adoc [hunk 13]
```diff
@@ -1,47 +1,78 @@
...
-Install Node.js and npm preferably using nvm as the Node Version Manager: link:https://github.com/nvm-sh/nvm/[https://github.com/nvm-sh/nvm]
+It is recommended to install Node.js and npm using the Node Version Manager (nvm):
+* NVM repository: [https://github.com/nvm-sh/nvm](https://github.com/nvm-sh/nvm)
++
+Alternatively, you can download and install Node.js from the official Node.js website:
+* Node.js official website: [https://nodejs.org](https://nodejs.org)
 +
-** Alternatively, download and install Node.js from the official Node.js website: link:https://nodejs.org/[https://nodejs.org]
-** After installing via either way, verify the installation by running the following commands in your terminal or command prompt:
+After installation, verify that both `node` and `npm` are correctly installed:
```
Comment (krisstern) [inline README.adoc:33]:
```suggestion
After the installation, verify that both `node` and `npm` are correctly installed:
```

File: README.adoc [hunk 14]
```diff
@@ -1,47 +1,78 @@
...
+----
+
+=== Installing Project Dependencies
+
+Once Node.js, npm, and Antora are installed, install the dependencies by running the following command at the root of the repository:
+
+[source,bash]  
 ----
-npm i
+npm install
```
Comment (krisstern) [inline README.adoc:56]:
```suggestion
npm i
```

Comment (krisstern) [reply README.adoc:56]: It can be either `npm i` or `npm install`, I don't need this needs to be changed.

File: README.adoc [hunk 15]
```diff
@@ -1,47 +1,78 @@
...
+[source,bash]  
 ----
 make all
 ----
-+
-Once the Antora site is up and running it should be available locally at link:http://127.0.0.1:8080/[http://127.0.0.1:8080]. 
+
+Once the site is up and running, it should be available locally at:
+
+* link:http://127.0.0.1:8080/[http://127.0.0.1:8080]
```
Comment (krisstern) [inline README.adoc:70]:
```suggestion
* link:http://127.0.0.1:8080/[http://127.0.0.1:8080]

```

**Review by krisstern — CHANGES_REQUESTED** [review]

File: README.adoc [hunk 16]
```diff
@@ -1,47 +1,79 @@
...
 ----
 node --version
 npm --version
 ----
 
-. **Intall dependencies**
-+
-Run the following at the root fo the repository:
+. **Install Antora CLI**
+
```
Comment (krisstern) [inline README.adoc:42]:
```suggestion
+
```

**Review by krisstern — CHANGES_REQUESTED** [review]
My bad, missed one

File: README.adoc [hunk 17]
```diff
@@ -1,47 +1,79 @@
...
+To build and serve the site locally, use the provided Makefile. Run this command at the root of the repository:
+
+[source,bash]  
 ----
 make all
 ----
-+
-Once the Antora site is up and running it should be available locally at link:http://127.0.0.1:8080/[http://127.0.0.1:8080]. 
+
```
Comment (krisstern) [inline README.adoc:67]:
```suggestion
+
```

Comment (krisstern) [reply README.adoc:67]: It looks like you are removing too many of these `+` signs that are needed for continuity.

**Review by krisstern — CHANGES_REQUESTED** [review]

**Comment by Brijeshthummar02** [comment]
@krisstern i have gone through whole file by my own and added + wherever needed, also mention if more changes are required.

File: README.adoc [hunk 18]
```diff
@@ -1,47 +1,80 @@
...
 ----
+
+Once the site is up and running, it should be available locally at:
+
+* link:http://127.0.0.1:8080/[http://127.0.0.1:8080]
+
+== Documentation Structure
+
+* **Antora Documentation**: This version-controlled documentation covers different guides (e.g., Developer Guide, User Guide).
 +
```
Comment (krisstern) [inline README.adoc:74]:
```suggestion

```

**Review by krisstern — CHANGES_REQUESTED** [review]

**Review by krisstern — APPROVED** [review]
Hi @Brijeshthummar02! LGTM, thanks for all the hard work

**Comment by Brijeshthummar02** [comment]
Hi @krisstern,

I just wanted to take a moment to thank you for all your support and guidance throughout this issue. Your patience and willingness to help have made a significant difference, and I truly appreciate it.

Best regards,
@Brijeshthummar02

**Comment by krisstern** [comment]
Hey you are welcome @Brijeshthummar02! It has been a pleasure working with you 😃

**Pull request merged**